core/variant/array.cpp

```diff
@@ -123,6 +123,10 @@ bool Array::recursive_equal(const Array &p_array, int recursion_count) const {
 	if (_p == p_array._p) {
 		return true;
 	}
+	if (_p->typed != p_array._p->typed) {
```

Comment:
Why would you not want this? I am not sure why you can't compare a typed array to untyped if the contents are the same.

Reply:
Remember that GDScript is by default untyped

Reply:
It is an error to compare `PackedInt32Array` and `PackedInt64Array` by value.

And there is an inconsistency about nested elements in array:
```gdscript
print('a' == &'a') # true
print(['a'] == [&'a']) # false
```

So it is an arbitrary choice, I guess. Have no problem removing this.

Reply:
I believe arrays of different types can be considered different, even if empty. But untyped array should be comparable with any typed or untyped array by its contents.

```gdscript
var typed: Array[int] = [1, 2, 3]
var untyped = [1, 2, 3]
print(typed == untyped) # true
```

Reply:
Related question: should type info be included in hash? So should `Array[StringName]()` and `Array[float]()` have the same hash or different one?

Reply:
Ok, I'll just remove my additions to `recursive_equal` and `recursive_hash`.